cmake/modules/antlr4cpp.cmake

```diff
@@ -2,13 +2,52 @@ message(STATUS "antlr4cpp - using bundled version")
 
 include(ExternalProject)
 
+# check if uuid-dev is installed on the system, since this is a dependency of antlr4cpp
+find_package(PkgConfig)
+pkg_check_modules(UUID QUIET uuid)
+
+# if uuid-dev was not found on system we install it locally
+if (NOT UUID_FOUND)
+    message(STATUS "UUID not found - using bundled version")
+
+    set(LIBUUID_INSTALL_DIR "${CMAKE_BINARY_DIR}/uuid/install")
+    set(LIBUUID_PKGCONFIG_DIR ${LIBUUID_INSTALL_DIR}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH})
+    set(LIBUUID_LIBRARY_DIR ${LIBUUID_INSTALL_DIR}/lib:$ENV{LIBRARY_PATH})
+    set(UUID_CONFIG_PARAMS "--prefix=${LIBUUID_INSTALL_DIR}")
```

Comment:
Do we have to set these Vars manually or can we use something like `find_package()`?

Reply:
The problem is that at this point `uuid` is not yet installed, i.e. `find_package()` doesn't find anything yet, but we need to set the variables so we can pass them to `antlr4cpp_bundled` in the next `ExternalProject_ADD`. 

Unfortunately, antlr4-cpp-runtime's CMakeLists.txt uses `pkg_check_modules()`, so we need to set the `PKG_CONFIG_PATH` and `LIBRARY_PATH` environment variables somehow.

An alternative, which would replace both setting the variables (like `LIBUUID_INSTALL_DIR`, `LIBUUID_PKGCONFIG_DIR`, ...) and the custom commands like 
- `CMAKE_COMMAND    ${CMAKE_COMMAND} -E env PKG_CONFIG_PATH=${LIBUUID_PKGCONFIG_DIR} ${CMAKE_COMMAND}` 
- `BUILD_COMMAND    ${CMAKE_COMMAND} -E env LIBRARY_PATH=${LIBUUID_LIBRARY_DIR} ${CMAKE_MAKE_PROGRAM}`)

would be to modify antlr4-cpp-runtime's CMakeLists.txt and instead of using `pkg_check_modules()` simply use a `find_library(UUID NAMES uuid uuid-dev libuuid HINTS "${CMAKE_BINARY_DIR}/../../../uuid/install/lib" REQUIRED)`. But of course this has the disadvantage that you have to adjust the CMakeLists.txt every time when upgrading antlr4cpp.

Reply:
Ok in that case leave it as it is